extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/context/VertxContextSafetyToggle.java

```diff
@@ -120,6 +120,10 @@ public static void setContextSafe(final Context context, final boolean safe) {
             throw new IllegalStateException(
                     "Can't set the context safety flag: the current context is not a duplicated context");
         } else {
+            // save storm of true -> true transitions by shielding it
+            if (safe && context.getLocal(ACCESS_TOGGLE_KEY) == Boolean.TRUE) {
```

Comment:
no need to shield for the case in which the `safe` parameter is false?

(not that I think it's used much)

Reply:
Given that doesn't seems to happen so often (in my test at least) I didn't add a path to handle that